src/components/engagement/engagement.service.ts

```diff
@@ -15,54 +26,823 @@ import {
+          canEditMethodologies.canEditMethodologies as canEditMethodologies,
+          canReadStatus.canReadStatus as canReadStatus,
+          canEditStatus.canEditStatus as canEditStatus,        
+          canReadCompleteDate.canReadCompleteDate as canReadCompleteDate,
+          canEditCompleteDate.canEditCompleteDate as canEditCompleteDate,
+          CanReadDisbursementCompleteDate.CanReadDisbursementCompleteDate as CanReadDisbursementCompleteDate,
+          CanEditDisbursementCompleteDate.CanEditDisbursementCompleteDate as CanEditDisbursementCompleteDate,
+          canReadCommunicationsCompleteDate.canReadCommunicationsCompleteDate as canReadCommunicationsCompleteDate,
+          canEditCommunicationsCompleteDate.canEditCommunicationsCompleteDate as canEditCommunicationsCompleteDate,
+          canReadStartDate.canReadStartDate as canReadStartDate,
+          canEditStartDate.canEditStartDate as canEditStartDate,
+          canReadEndDate.canReadEndDate as canReadEndDate,
+          canEditEndDate.canEditEndDate as canEditEndDate,
+          canReadInitialEndDate.canReadInitialEndDate as canReadInitialEndDate,
+          canReadLastSuspendedAt.canReadLastSuspendedAt as canReadLastSuspendedAt,
+          canReadLastReactivatedAt.canReadLastReactivatedAt as canReadLastReactivatedAt,
+          canReadStatusModifiedAt.canReadStatusModifiedAt as canReadStatusModifiedAt,
+          canReadModifiedAt.canReadModifiedAt as canReadModifiedAt
+      `;
+    }
+    const result = await this.db
+      .query()
+      .raw(query, {
+        token: session.token,
+        requestingUserId: session.userId,
+        owningOrgId: session.owningOrgId,
+        id,
+      })
+      .first();
+
+    if (!result) {
+      throw new NotFoundException(
+        'Could not find langage engagement or internship engagement'
+      );
+    }
+
+    const ceremony = await this.ceremonyService.readOne(
+      result.ceremonyId,
+      session
+    );
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    let language: Language = {} as Language;
+    if (result.languageId) {
+      language = await this.languageService.readOne(result.languageId, session);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    let internUser: User = {} as User;
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    let mentorUser: User = {} as User;
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    let countryOfOrigin: Country = {} as Country;
+    if (result.internUserId) {
+      internUser = await this.userService.readOne(result.internUserId, session);
+    }
+    if (result.mentorUserId) {
+      mentorUser = await this.userService.readOne(result.mentorUserId, session);
+    }
+    if (result.countryOfOriginId) {
+      countryOfOrigin = (await this.locationService.readOne(
+        result.countryOfOriginId,
+        session
+      )) as Country;
```

Comment:
Looks like `LocationService` has a `readOneCountry` which would remove the need to typecast.